pkg/helm/handlers/handler_test.go

```diff
@@ -9,19 +12,100 @@ import (
-	t.Run("List helm releases", func(t *testing.T) {
-		fakeHandlers := HelmHandlers{
-			GetActionConfigurations: fake.GetActionConfigurations,
-			RenderManifests:         fake.RenderManifests,
-			InstallChart:            fake.InstallChart,
-			ListReleases:            fake.ListReleases,
-		}
+	tests := []struct {
+		test        string
+		expectedMsg string
+	}{
+		{test: "invalid test", expectedMsg: "{\"error\":\"Failed to list helm releases: unknown error occurred\"}"},
+		{test: "valid test", expectedMsg: "[{\"name\":\"Test\"}]"},
+	}
+	for _, tt := range tests {
+		t.Run("List helm releases", func(t *testing.T) {
+			handlers := FakeHelmHandler()
+			if tt.test == "valid test" {
+				handlers.ListReleases = fake.ListReleases
+			} else if tt.test == "invalid test" {
+				handlers.ListReleases = fake.ListReleasesFailed
+			}
+
+			request := httptest.NewRequest("", "/foo", strings.NewReader("{}"))
+			response := httptest.NewRecorder()
+
+			handlers.HandleHelmList(&auth.User{}, response, request)
+			switch tt.test {
+			case "valid test":
+				if response.Code != http.StatusOK {
+					t.Error("Failed to install release")
+				}
+				if bytes.Compare(response.Body.Bytes(), []byte(tt.expectedMsg)) != 0 {
+					t.Error("response body not matching")
+				}
+			case "invalid test":
+				if response.Code != http.StatusBadRequest {
+					t.Error("response code should be 400")
+				}
+				if bytes.Compare(response.Body.Bytes(), []byte(tt.expectedMsg)) != 0 {
+					t.Error("response body not matching")
+				}
+			}
+		})
+	}
+}
 
-		request := httptest.NewRequest("", "/foo", strings.NewReader("{}"))
-		response := httptest.NewRecorder()
+func TestHelmHandlers_HandleHelmInstall(t *testing.T) {
+	tests := []struct {
+		test        string
+		expectedMsg string
+	}{
+		{test: "invalid test", expectedMsg: "{\"error\":\"Failed to install helm chart: Chart path is invalid\"}"},
+		{test: "valid test", expectedMsg: "{\"name\":\"Test\"}"},
+	}
+	for _, tt := range tests {
+		t.Run("Install Helm release", func(t *testing.T) {
+			handlers := FakeHelmHandler()
+			if tt.test == "valid test" {
+				handlers.InstallChart = fake.InstallChart
+			} else if tt.test == "invalid test" {
+				handlers.InstallChart = fake.InstallChartFailed
+			}
 
-		fakeHandlers.HandleHelmList(&auth.User{}, response, request)
+			req := HelmRequest{
+				Name:      "test",
+				Namespace: "test",
+				ChartUrl:  "../testdata/influxdb-3.0.2.tgz",
+				Values:    nil,
+			}
+			body, err := json.Marshal(req)
+			if err != nil {
+				t.Fail()
+			}
+			request := httptest.NewRequest("", "/foo", bytes.NewReader(body))
+			response := httptest.NewRecorder()
 
-	})
+			handlers.HandleHelmInstall(&auth.User{}, response, request)
+			switch tt.test {
+			case "valid test":
+				if response.Code != http.StatusOK {
+					t.Error("Failed to install release")
+				}
+				if bytes.Compare(response.Body.Bytes(), []byte(tt.expectedMsg)) != 0 {
+					t.Error("response body not matching")
```

Comment:
I would display here what was expected and what was received - it would help to spot the issue faster.